apps/website/src/components/courses/exercises/MultipleChoice.tsx

```diff
@@ -48,93 +47,135 @@ const MultipleChoice: React.FC<MultipleChoiceProps> = ({
-    setSelectedOption(option);
     setValue('answer', option);
     setIsEditing(true);
   };
 
-  const isSelected = (option: string): boolean => {
-    if (!isEditing && formattedExerciseResponse) {
-      return formattedExerciseResponse === option;
-    }
-    return selectedOption === option;
-  };
-
   useEffect(() => {
     if (formattedExerciseResponse) {
       setValue('answer', formattedExerciseResponse);
     }
   }, [formattedExerciseResponse, setValue]);
 
-  const onSubmit = useCallback(async (data: FormData) => {
-    const isAnswerCorrect = data.answer === formattedAnswer;
-    await onExerciseSubmit(data.answer, isAnswerCorrect);
-    setIsEditing(false);
-  }, [onExerciseSubmit]);
+  const onSubmit = useCallback(
+    async (data: FormData) => {
+      const isAnswerCorrect = data.answer === formattedAnswer;
+      await onExerciseSubmit(data.answer, isAnswerCorrect);
+      setIsEditing(false);
+    },
+    [onExerciseSubmit],
+  );
 
-  const isCorrect = formattedExerciseResponse && formattedExerciseResponse === formattedAnswer;
-  const isIncorrect = formattedExerciseResponse && formattedExerciseResponse !== formattedAnswer;
+  const handleTryAgain = () => {
+    // Don't block on this request, we want to reset state immediately
+    onExerciseSubmit('', false);
+    setIsEditing(true);
+    setValue('answer', '');
+  };
+
+  const getSubmitButtonText = () => {
+    if (isSubmitting) return 'Checking...';
+    if (currentAnswer) return 'Check answer';
+    return 'Select an option'; // No quiz options have been selected yet
+  };
+
+  const isCorrect = !isEditing && formattedExerciseResponse && formattedExerciseResponse === formattedAnswer;
+  const isIncorrect = !isEditing && formattedExerciseResponse && formattedExerciseResponse !== formattedAnswer;
+
+  const getOptionClasses = (option: string) => {
+    const selected = currentAnswer === option;
+
+    if (!selected) {
+      // If there is a submitted answer, or the user is not logged in, dim unselected option text and don't allow hover effects.
+      return `bg-[#2A2D340A] border-transparent ${isCorrect || isIncorrect || !isLoggedIn ? 'text-gray-400' : 'hover:bg-[#F0F5FD]'}`;
+    }
+    if (isCorrect) return 'bg-[#18B71B1A] border-[#18B71B]';
+    if (isIncorrect) return 'bg-[#DC00001A] border-[#DC0000]';
+
+    // Default style for selected option (when no answer has been submitted yet)
+    return 'bg-[#F0F5FD] border-[#2244BB]';
+  };
 
   return (
-    <form onSubmit={handleSubmit(onSubmit)} className={clsx('multiple-choice container-lined bg-white p-8 flex flex-col gap-6', className)}>
-      <div className="multiple-choice__header flex flex-col gap-4">
-        <div className="multiple-choice__header-content flex flex-col gap-2">
-          <p className="multiple-choice__title bluedot-h4 not-prose">{title}</p>
-          <MarkdownExtendedRenderer className="multiple-choice__description">{description}</MarkdownExtendedRenderer>
+    <div>
```

Comment:
I needed to wrap the form in a container div so that the 'Correct!' message could be displayed underneath the form.